django/db/backends/sqlite3/introspection.py

```diff
@@ -280,14 +366,33 @@ def get_constraints(self, cursor, table_name):
             # table_name is a view.
             pass
         else:
-            constraints.update(self._parse_table_constraints(table_schema))
+            columns = {info.name for info in self.get_table_description(cursor, table_name)}
+            constraints.update(self._parse_table_constraints(table_schema, columns))
 
         # Get the index info
         cursor.execute("PRAGMA index_list(%s)" % self.connection.ops.quote_name(table_name))
         for row in cursor.fetchall():
             # SQLite 3.8.9+ has 5 columns, however older versions only give 3
             # columns. Discard last 2 columns if there.
             number, index, unique = row[:3]
+
+            cursor.execute(
+                "SELECT sql FROM sqlite_master "
+                "WHERE type='index' AND name=%s" % self.connection.ops.quote_name(index)
+            )
+            # There would be only 1 row
+            sql, = cursor.fetchone() or (None,)
+            # Inline constraints already detected in _parse_table_constraints
+            # The reasons to avoid inline constraints in `PRAGMA index_list`:
+            # - inline constraints can have noname or different to `PRAGMA index_list` name
+            #   so you should avoid duplicates for it
+            # - two and more inline constraints can have only one index in `index_list` for same columns
+            # - inline constraints have definition within table (no `CREATE INDEX` definition)
```

Comment:
I'm not sure if I understand this comment. Is this an accurate rewriting or have I missed the point?
```
# - PRAGMA index_list doesn't provide any additional information
#   about inline constraints.
```

Reply:
I think no, just try explain on next example:

```
sqlite> CREATE TABLE "test2" (
   "id" integer NOT NULL PRIMARY KEY AUTOINCREMENT,
   "name" varchar(255) NOT NULL UNIQUE,
    CONSTRAINT test1_uniq UNIQUE (name)
);

sqlite> PRAGMA index_list('test2');
0|sqlite_autoindex_test2_1|1|u|0

sqlite>  PRAGMA index_info('sqlite_autoindex_test2_1');
0|1|name

sqlite> SELECT * FROM sqlite_master;
table|test2|test2|5|CREATE TABLE "test2" ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "name" varchar(255) NOT NULL UNIQUE, CONSTRAINT test1_uniq UNIQUE (name))
index|sqlite_autoindex_test2_1|test2|6|
```

- you can find that index name in `PRAGMA index_list` different to inline definition (`test1_uniq` vs `sqlite_autoindex_test2_1`)
- there are two constraints defined inline, but only one index visible in  `PRAGMA index_list`
- and there are empty definition in `SELECT * FROM sqlite_master` for `sqlite_autoindex_test2_1` index

eg  `PRAGMA index_list` provide information about indexes but this information different to constraints definition.

Can you help me to describe it more clearly or just left information that index information in `PRAGMA` different to constraint definition?

Reply:
I pushed a commit with some edits, let me know if it looks okay.

Reply:
Looks fine